src/ruptures/metrics/randindex.py

```diff
@@ -14,3 +15,41 @@ def randindex(bkps1, bkps2):
         float: Rand index
     """
     return 1 - hamming(bkps1, bkps2)
+
+
+def randindex_cpd(bkps1, bkps2):
+    """Computes efficiently the Rand index for change-point detection given two
+    sorted partitions.
+
+    Args:
+        bkps1 (list): sorted list of the last index of each regime.
+        bkps2 (list): sorted list of the last index of each regime.
+
+    Returns:
+        float: Rand index
+    """
+    sanity_check(bkps1, bkps2)
+    n_samples = max(bkps1)
+    disagreement = 0
+    bkps1.insert(0, 0)
+    bkps2.insert(0, 0)
+
+    beginj = 0  # avoids unnecessary computations
+    for i in range(len(bkps1) - 1):
```

Comment:
Try to use more informative variable names (not `i` or `j`)
```suggestion
    for (start1, end1) in pairwise([0]+bkps1):
```

(`pairwise` is available in `ruptures.utils`.)